azurelinuxagent/common/protocol/wire.py

```diff
@@ -873,34 +926,61 @@ def get_certs(self):
         return self.certs
 
-    def get_current_handlers(self):
-        handler_list = list()
-        try:
-            incarnation = self.fetch_cache(os.path.join(conf.get_lib_dir(),
-                                                        INCARNATION_FILE_NAME))
-            ext_conf = ExtensionsConfig(self.fetch_cache(os.path.join(conf.get_lib_dir(),
-                                                                      EXT_CONF_FILE_NAME.format(incarnation))))
-            handler_list = ext_conf.ext_handlers.extHandlers
-        except ProtocolError as pe:
-            # cache file is missing, nothing to do
-            logger.verbose(ustr(pe))
-        except Exception as e:
-            logger.error("Could not obtain current handlers: {0}", ustr(e))
-
-        return handler_list
+    def get_sequence_number(self):
+        path = os.path.join(conf.get_lib_dir(), SEQUENCE_NUMBER_FILE_NAME)
+        if os.path.exists(path):
+            return int(fileutil.read_file(path))
+        else:
+            return 0
 
-    def get_ext_conf(self):
-        if self.ext_conf is None:
-            goal_state = self.get_goal_state()
-            if goal_state.ext_uri is None:
+    def get_use_fast_track(self):
+        path = os.path.join(conf.get_lib_dir(), GOAL_STATE_SOURCE_FILE_NAME)
+        if os.path.exists(path):
+            goal_state_source = fileutil.read_file(path)
+            if goal_state_source == GOAL_STATE_SOURCE_FASTTRACK:
+                return True
+        else:
+            return False
+
+    def get_ext_conf(self, read_always=False):
+        if self.ext_conf is None or read_always:
+            goal_state = None
+            local_file = None
+            use_fast_track = self.get_use_fast_track()
+            if use_fast_track:
+                local_file = self.get_file_path_for_fast_track()
+            if local_file is None:
+                local_file = self.get_file_path_for_fabric(goal_state)
+            if local_file is None:
+                logger.info("No extensions file to read. Using empty ExtensionsConfig")
                 self.ext_conf = ExtensionsConfig(None)
             else:
-                local_file = EXT_CONF_FILE_NAME.format(goal_state.incarnation)
-                local_file = os.path.join(conf.get_lib_dir(), local_file)
                 xml_text = self.fetch_cache(local_file)
                 self.ext_conf = ExtensionsConfig(xml_text)
         return self.ext_conf
 
+    def get_file_path_for_fabric(self, goal_state):
+        if goal_state is None:
+            goal_state = self.get_goal_state()
+        if goal_state.ext_uri is None:
+            return None
+        local_file = EXT_CONF_FILE_NAME.format(goal_state.incarnation)
+        local_file = os.path.join(conf.get_lib_dir(), local_file)
+        return local_file
+
+    def get_file_path_for_fast_track(self):
+        local_file = None
+        sequence_number = self.get_sequence_number()
+        if sequence_number > 0:
```

Comment:
Not sure how the extension config sequence number works, but can it ever be 0? If so then maybe you can change the default in function `get_sequence_number` to a negative number?